src/Symfony/Component/PropertyInfo/Tests/Extractor/PhpDocExtractorTest.php

```diff
@@ -548,6 +548,7 @@ public static function typeProvider(): iterable
         yield ['collection', Type::list(Type::object(\DateTimeImmutable::class)), null, null];
         yield ['nestedCollection', Type::list(Type::list(Type::string())), null, null];
         yield ['mixedCollection', Type::array(), null, null];
+        yield ['nullableTypedCollection', Type::nullable(Type::list(Type::object(Dummy::class))), null, null];
```

Comment:
we should expan `provideLegacyTypes()` in the same way to ensure that the behaviour of old and new type sytem matches

Reply:
Done, amended to my last commit